### DISCUSSION TITLE
Tapo Two Way Audio Not Working

### DISCUSSION THREAD

**Discussion by sfnemis** [koush/scrypted#1363] [General]
Hi,

I have various models of Tapo cameras. I've integrated most of them with Scrypted, especially using the onvif plugin, Tapo camera plugin, and homekit plugins. I even purchased an NVR license and set up my cameras, and everything works very well. However, I absolutely cannot use the two-way intercom feature with the microphone through the Apple Home app. I've been struggling for about 2 days, reading Reddit comments, and even following GitHub issues, but I think there is no solution. Everything works perfectly; I just wish the two-way audio feature could work smoothly as well :( Additionally, it's interesting that we can receive audio from the camera, but when we use the "talk" feature in the Home app, our voice isn't transmitted to the cameras. There's something odd going on here, but unfortunately, I haven't been able to resolve it :(

If anyone has any idea please share here.

**Comment by jvic1234** [comment]
You need to use the tapo plugin with the onvif plugin. Certain tapo cameras don't work with "way audio. Please check on discord if you require further assistance

**Reply by sfnemis** [reply]
Hi, 

I know already i installed with onvif plugin too ! It's not working at all kind of Tapo Cameras.

**Reply by radityakurnianto** [reply]
are you using this plugin? https://www.npmjs.com/package/@scrypted/tapo?activeTab=readme

**Reply by sfnemis** [reply]
Thank you ! For sure i installed the plugin

[Tapo Two Way Audio]:   Stream #0:0: Audio: opus, 48000 Hz, stereo, fltp
[Tapo Two Way Audio]: Stream mapping:
[Tapo Two Way Audio]:   Stream #0:0 -> #0:0 (opus (native) -> pcm_alaw (native))
[Tapo Two Way Audio]: Press [q] to stop, [?] for help
[Tapo Two Way Audio]: Output #0, rtp, to 'rtp://127.0.0.1:43671?':
[Tapo Two Way Audio]:   Metadata:
[Tapo Two Way Audio]:     encoder         :
[Tapo Two Way Audio]: Lavf58.76.100
[Tapo Two Way Audio]:   Stream #0:0: Audio: pcm_alaw, 8000 Hz, mono, s16, 64 kb/s
[Tapo Two Way Audio]:     Metadata:
[Tapo Two Way Audio]:       encoder         : Lavc58.134.100 pcm_alaw
[Tapo Two Way Audio]: size=       0kB time=00:00:00.00 bitrate=N/A speed=N/A
[Tapo Two Way Audio]: video/audio detected, discarding further input
[WebRTC Plugin]: signalingState closed
[WebRTC Plugin]: connectionState closed
[WebRTC Plugin]: iceConnectionState closed
[Rebroadcast Plugin]: mainStream active rebroadcast clients: 0
[Tapo Two Way Audio]: ffmpeg exited

**Reply by radityakurnianto** [reply]
I recently add that plugin for mya tapo c200 and it works. You need to enter your tapo app account password (not the password for the RTSP) 


![image](https://github.com/koush/scrypted/assets/35289964/c407540d-f210-4b35-bfa4-fdde0c8b0577) 


Also don't forget to check this inside the tapo two way audio plugin

<img width="902" alt="image" src="https://github.com/koush/scrypted/assets/35289964/ca30b05d-3dc0-4983-8d5a-a9b1a3e94c6a">

**Reply by sfnemis** [reply]
Hi, 

Thank you for all your details, but i already done all of them i have a lot of Tapo Models Cam yes its working just one of C210 Model camera the another models like Tapo C520WS, Tapo C325WB, Tapo C500, Tapo C510W none of them not working !

**Reply by gmalhotra02** [reply]
I am also struggling for Two Way Audio & PTZ, couldn't get it work with any tapo camera - followed all steps 

I have C225 & C210 cameras

- Have configured Tapo Camera with OVNIF Plugin
- Installed and configured Tapo Camera
- Enabled Required Extensions
- Added Password for Two Way Audio
- Enabled PTZ for PTZ as well

I see the mic button on Home Kit however two way audio doesn't work, 

PTZ (Pan, Tilt & Zoom)  accessory is not available. 

Is there any IP settings to be done?

**Reply by tellok** [reply]
> tapo c200

does c120 support two way audio in scrypted using @scrypted/tapo?

edit:
yes, it's supported, my firmware version is v1.1.15

**Comment by koush** [comment]
two way audio was just fixed for a variety of tape in 0.0.13

**Reply by gmalhotra02** [reply]
With 0.0.13 my tapo c225 is working however c210 is not working. My C210 is working with 0.0.11 however then c225 is not working with 0.0.11 

for me among all versions tested either of the one is working with 0.0.11 or 0.0.13 

no version is able to support both cameras.

rest of the versions I tested, doesn’t support any of them 

note I am on latest Sycypted version 0.102.0

**Comment by benjoon90** [comment]
Hi I too have a C225 which I can’t get the two way audio to work .. All other function worked fine.

I’ve added the cam to Onvif plugin, added the Tapo Plugin as well as the Two Way Audio plugin.
Enabled it for the cam and went to the Cam setting on Onvif plugin to add my cloud password. 
Still no voice coming out from camera. 

Using the Tapo app have no problems though.

**Comment by radityakurnianto** [comment]
I can say that it is not really stable


*Raditya Kurnianto**iOS Developer*


*linked.in <https://id.linkedin.com/in/radityakurnianto>*

**Comment by DracoC77** [comment]
I have been unable to get my C120 v1.6HW working two way audio despite trying all different combinations of plugin versions and making sure two way audio work with the app. I suspect that Tapo has been rolling our security changes in the last year that breaks the previous two way audio implementation because I'm also unable to get go2rtc tapo 2 way audio working either (which return a 401 error regardless of the password method):

Possibly reated issue:
https://github.com/AlexxIT/go2rtc/issues/849
